lib/registries/addon/components/sharing-icons/dropdown/component.ts

```diff
@@ -0,0 +1,70 @@
+import { tagName } from '@ember-decorators/component';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import { task } from 'ember-concurrency';
+import { DS } from 'ember-data';
+import I18n from 'ember-i18n/services/i18n';
+import Toast from 'ember-toastr/services/toast';
+
+import { layout } from 'ember-osf-web/decorators/component';
+import Collection from 'ember-osf-web/models/collection';
+import Registration from 'ember-osf-web/models/registration';
+import styles from './styles';
+import template from './template';
+
+@tagName('')
+@layout(template, styles)
+export default class SharingIconsDropdown extends Component.extend({
+    bookmark: task(function *(this: SharingIconsDropdown) {
+        if (!this.bookmarksCollection || !this.node) {
+            return;
+        }
+
+        const op = this.isBookmarked ? 'remove' : 'add';
+        let response;
+
+        try {
+            if (op === 'remove') {
+                this.bookmarksCollection.linkedRegistrations.removeObject(this.node);
+                response = yield this.bookmarksCollection.deleteM2MRelationship('linkedRegistrations', this.node);
+            } else {
+                this.bookmarksCollection.linkedRegistrations.pushObject(this.node);
+                response = yield this.bookmarksCollection.createM2MRelationship('linkedRegistrations', this.node);
```

Comment:
I get an error trying to add a bookmark -- looks like the API expects a `POST` instead of a `CREATE` request